module/VuFind/src/VuFind/GeoFeatures/BasemapConfig.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * Basemap Configuration Module
+ *
+ * PHP version 5
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  GeoFeatures
+ * @author   Leila Gonzales <lmg@agiweb.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:recommendation_modules Wiki
+ */
+namespace VuFind\GeoFeatures;
+
+use Zend\ServiceManager\ServiceManager;
+
+/**
+ * Basemap Configuration Class
+ *
+ * @category VuFind
+ * @package  GeoFeatures
+ * @author   Leila Gonzales <lmg@agiweb.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:hierarchy_components Wiki
+ */
+class BasemapConfig
+{
+    /**
+     * Basemap tileserver URL
+     * Default is the wikimedia osm-intl map
+     *
+     * @var string
+     */
+    protected $basemap_url = "https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}.png";
+
+    /**
+     * Basemap attribution
+     *
+     * @var string
+     */
+    protected $basemap_attribution = '<a href="https://wikimediafoundation.org/wiki/
+        Maps_Terms_of_Use">Wikimedia</a> | © <a href="https://www.openstreetmap.org/
+        copyright">OpenStreetMap</a>';
+
+    /**
+     * Request origin
+     *
+     * @var string
+     */
+    protected $requestOrigin;
+
+    /**
+     * Get the basemap configuration settings.
+     *
+     * @param ServiceManager $sm     Service manager.
```

Comment:
Rather than using the service manager as a parameter to these functions, I'd instead recommend injecting the VuFind\Config service as a dependency through the constructor (and using a factory to build the class). It's generally a good idea to keep the service manager as far away from your classes as possible, since that makes the code easier to maintain -- the service manager can potentially offer access to an infinite variety of unrelated things. For the purposes of understanding relationships between classes, it's better to pass in more specific things so we can figure out where the dependencies are.

Reply:
Good point. I hadn't considered that. Thanks for the feedback. I'll re-write this to use a factory to build the class then.

Reply:
I see that you've added the factory, but it does not appear that the matching changes to this class were committed. I don't see a constructor to accept the config manager or the removal of $sm parameters to all the methods.